content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+| ON  | OFF | DX12              | Each SRG can contain one last unbounded array for each: <br> t[], u[], b[], s[]|
+| ON  | ON  | Vulkan<br> Metal  | Each SRG can contain one, and only one, last unbounded array across all resource types: <br> Only one of t[], u[], b[], s[]|
+  
+## Things to know before going over some examples:
+* "Register Space" in HLSL is the same as "Descriptor Set" in Vulkan.
+* When "--use-spaces" is used, **AZSLc** will assign a unique Register Space per SRG.
+* When "--unique-idx" is used, **AZSLc** will emit all resource descriptors in a Register Space to be in numerical sequence, regardless of the descriptor type. Platforms like Vulkan and Metal require this.
+* Unbounded arrays take over the whole range of register indices from the point they are being declared. This means two things:
+    1. It is possible to declare any kind of resources or any amount of bounded arrays of resources, BEFORE any given Unbounded Array of a particular kind.
+    2. Any resource declared AFTER an Unbounded Array will be considered an error.
+  
+## Examples
+All the examples shown here assume '--use-spaces' is ON when compiling the shader. Which means each SRG will be assigned to its own Register Space.
+
+### Example 1
+In this example there's only one SRG, and no unbounded arrays yet. It is useful to understand register space & index binding assignment.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t1 register.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
+```
+### Example 2: (Error case)
+This example is based on Example 1, but this time We are trying to introduce the first Unbounded Array: `Texture2D<float4> m_texSRV_unbounded[]`.  
+This example will fail to compile because, as mentioned before, an Unbounded Array takes over the whole register starting from their index of declaration.  
+In this example, `m_texSRV_unbounded`, is the first texture SRV so it will start at **t0**, and take over the rest of the indices until **tN**. Any other texture SRV declared afterwards won't find a register index to bind to.  
```

Comment:
```suggestion
This example fails to compile because `m_texSRV_unbounded[]` takes over the whole range of register indices from the point its declared. The unbounded array, `m_texSRV_unbounded[]`, is the first texture shader resource view (SRV), so it takes over the whole range of register indices that's allocated for SRVs, from `t0` to `tN`. Texture SRVs that are declared afterwards don't have a register index to bind to.  
```
* Use in-line code style for code elements like `t[ ]` or `tN`. 
* Write out "Shader Resource View (SRV)". For all acronyms across all pages, first introduce the full term and its acronym, then use just the acronym. This is very helpful to readers.

Reply:
Fixed.